src/group_theory/perm/support.lean

```diff
@@ -172,16 +172,42 @@ variable [decidable_eq α]
 /-- `f.is_swap` indicates that the permutation `f` is a transposition of two elements. -/
 def is_swap (f : perm α) : Prop := ∃ x y, x ≠ y ∧ f = swap x y
 
+@[simp] lemma of_subtype_swap_eq {p : α → Prop} [decidable_pred p]
+  (x y : subtype p) : -- (hxy : x ≠ y) :
+  (equiv.swap x y).of_subtype = equiv.swap ↑x ↑y :=
+equiv.ext $ λ z, begin
+  by_cases hz : p z,
```

Comment:
I like this proof, and propose to keep it. I feel like "symmetric" lines are closer together in this version.

Reply:
OK. I just pushed a new version where the old commented proof has been simply deleted.